pkg/log/checkpoint/dns_test.go

```diff
@@ -0,0 +1,188 @@
+package checkpoint
+
+import (
+	"context"
+	"errors"
+	"net"
+	"reflect"
+	"testing"
+
+	"github.com/kolide/launcher/pkg/log/checkpoint/mocks"
+)
+
+func Test_lookupHostsIpv4s(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		ipLookuper *mocks.IpLookuper
+		hosts      []string
+	}
+	tests := []struct {
+		name              string
+		args              args
+		onLookupIPReturns []func() ([]net.IP, error)
+		want              []string
+	}{
+		{
+			name: "happy_path_single",
+			args: args{
+				ipLookuper: &mocks.IpLookuper{},
+				hosts:      []string{"happy_path_single.com"},
+			},
+			onLookupIPReturns: []func() ([]net.IP, error){
+				func() ([]net.IP, error) {
+					return []net.IP{
+
+						net.ParseIP("192.0.0.0"),
+						net.ParseIP("2001:db8::68"),
+					}, nil
+				},
+			},
+			want: []string{"happy_path_single.com: 192.0.0.0"},
```

Comment:
I always feel weird when we pick things outside of `example.com` but it seems okay in tests. (Less okay when we use `acme.com` in docs)

Reply:
that will work, I can start using that, thank you